### PR TITLE
Only load the Facebook SDK if we have a share action on the page.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?

This pull request implements one of my dreams - only loading the flipping huge [Facebook SDK](https://developers.facebook.com/docs/javascript) (215kb, 67kb gzipped) if we actually need it. I've also added a "simplified" share dialog that we use for other shares, which works identically but does not provide `completed` or `cancelled` Puck events.

Finally, I made a few little touch-ups to the `showFacebookShareDialog` helper, with [a new Promise-ified API](https://user-images.githubusercontent.com/583202/39266444-0588c5be-4898-11e8-9799-8216768f876f.png) & removed the need for `ShareActionContainer` by removing an unused prop & replacing `PuckConnector` with the new standalone helper functions.

### Any background context you want to provide?
My original dream was to load the SDK on-demand when a user clicked the "Share on Facebook" button. Unfortunately this dream was squashed by the way browser's pop-up blocks are implemented - calls to `window.open` only work if they were initiated by a user click event. Once we "defer" to make a network request, it's no longer a "user-initiated" action and is blocked. 🙅‍♂️

### What are the relevant tickets/cards?

[Discussion #17](https://github.com/orgs/DoSomething/teams/team-rocket/discussions/17)

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/Share/ShareContainer.js [hunk 1]
```diff
@@ -1,10 +1,5 @@
...
 import { PuckConnector } from '@dosomething/puck-client';
 import Share from './Share';
 
-const mapStateToProps = state => ({
-  user: state.user,
-});
-
-// Export the container component.
-export default connect(mapStateToProps)(PuckConnector(Share));
+// @TODO: Remove PuckConnector.
```
Comment (weerd) [inline resources/assets/components/Share/ShareContainer.js]: 🎉

File: resources/assets/components/actions/ShareAction/ShareAction.js [hunk 2]
```diff
@@ -3,40 +3,43 @@ import PropTypes from 'prop-types';
...
-
-    showFacebookSharePrompt({ href: link }, response => {
-      if (!response) {
-        trackEvent('share action cancelled', trackingData);
-        return;
-      }
+  componentDidMount() {
+    // If this is a Facebook share action, make sure we load SDK.
+    if (this.props.socialPlatform === 'facebook') {
+      loadFacebookSDK();
```
Comment (weerd) [inline resources/assets/components/actions/ShareAction/ShareAction.js]: yaaaaas!

File: resources/assets/helpers/index.js [hunk 3]
```diff
@@ -451,6 +469,31 @@ export function showTwitterSharePrompt(href, quote = '', callback) {
   }
 }
 
+/**
+ * Share a link by generating a Twitter intent share prompt.
```
Comment (weerd) [inline resources/assets/helpers/index.js]: typo: should be "generating a Facebook intent..."

Comment (DFurnes) [reply resources/assets/helpers/index.js]: Fixed in dd7121e! 👀

**Pull request merged**